stdlib/2and3/mmap.pyi

```diff
@@ -71,6 +70,9 @@ if sys.version_info >= (3,):
         def __setitem__(self, index: int, object: int) -> None: ...
         @overload
         def __setitem__(self, index: slice, object: bytes) -> None: ...
+        # Doesn't actually exist, but the object is actually iterable because it has __getitem__ and
+        # __len__, so we claim that there is also an __iter__ to help type checkers.
```

Comment:
Thanks! The only problem I see with that approach is that `CompatIterable` isn't in typing, so realistically most places where people really should use `CompatIterable` will actually use `Iterable`. We could add `CompatIterable` to typing but I feel like the complexity isn't worth it.

I'm personally OK with using the solution from this PR (adding `__iter__` even though it doesn't actually exist) more widely. We're already doing that for `str` in Python 2: it doesn't have `__iter__` but typeshed claims it does.